build_docker_image.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Description:
+# bash build_docker_image.sh MODE=stable
+# bash build_docker_image.sh MODE=nightly
+# bash build_docker_image.sh MODE=head
+
+# bash build_docker_image.sh MODE=stable
+# bash build_docker_image.sh MODE=stable JAX_VERSION=0.4.13
+# bash build_docker_image.sh MODE=stable JAX_VERSION=0.4.13 IMAGE_PREFIX=jax0.4.13
+
+
+# Enable "exit immediately if any command fails" option
+set -e
+
+export PROJECT=$(gcloud config get-value project)
+export IMAGE_PREFIX=gke_img
+
+# Set environment variables
+for ARGUMENT in "$@"; do
+    IFS='=' read -r KEY VALUE <<< "$ARGUMENT"
+    export "$KEY"="$VALUE"
+    echo "$KEY"="$VALUE"
+done
+echo "PROJECT=$PROJECT"
+
+if [[ "$MODE" == "stable" || ! -v MODE ]]; then
+    # Stable mode
+  if [[ ! -v JAX_VERSION ]]; then
+    docker build --build-arg MODE="stable" -f ./maxtext.Dockerfile -t ${IMAGE_PREFIX}_maxtext_stable .
+    docker tag ${IMAGE_PREFIX}_maxtext_stable gcr.io/$PROJECT/${IMAGE_PREFIX}_maxtext_stable:latest
```

Comment:
Nit: Not a blocker but if these two lines are the same as the else, they can be pulled outside of this if/else

Reply:
Good call!

Done.